src/toast.tsx

```diff
@@ -1,366 +1,368 @@
+    const panReleaseToRight = (gestureState: PanResponderGestureState) => {
+        Animated.timing(getPanResponderAnim(), {
+            toValue: { x: (dims.width / 10) * 9, y: gestureState.dy },
             useNativeDriver: Platform.OS !== "web",
-          }).start();
+            duration: 250,
+        }).start(() => onDestroy());
+    };
+
+    const getPanResponder = () => {
+        if (panResponderRef.current) return panResponderRef.current;
+        panResponderRef.current = PanResponder.create({
+            onMoveShouldSetPanResponder: (_, gestureState) => {
+                //return true if user is swiping, return false if it's a single click
+                return !(gestureState.dx === 0 && gestureState.dy === 0);
+            },
+            onPanResponderMove: (_, gestureState) => {
+                getPanResponderAnim()?.setValue({
+                    x: gestureState.dx,
+                    y: gestureState.dy,
+                });
+            },
+            onPanResponderRelease: (_, gestureState) => {
+                if (gestureState.dx > 50) {
+                    panReleaseToRight(gestureState);
+                } else if (gestureState.dx < -50) {
+                    panReleaseToLeft(gestureState);
+                } else {
+                    Animated.spring(getPanResponderAnim(), {
+                    toValue: { x: 0, y: 0 },
+                    useNativeDriver: Platform.OS !== "web",
+                }).start();
+                }
+            },
+        });
+        return panResponderRef.current;
+    };
+
+    const getPanResponderAnim = () => {
+        if (panResponderAnimRef.current) return panResponderAnimRef.current;
+        panResponderAnimRef.current = new Animated.ValueXY({ x: 0, y: 0 });
+        return panResponderAnimRef.current;
+    };
+
+    if (icon === undefined) {
+        switch (type) {
+            case "success": {
+                if (successIcon) {
+                    icon = successIcon;
+                }
+                break;
+            }
+
+            case "danger": {
+                if (dangerIcon) {
+                    icon = dangerIcon;
+                }
+                break;
+            }
+            case "warning": {
+                if (warningIcon) {
+                    icon = warningIcon;
+                }
+                break;
+            }
         }
-      },
-    });
-    return panResponderRef.current;
-  };
-
-  const getPanResponderAnim = () => {
-    if (panResponderAnimRef.current) return panResponderAnimRef.current;
-    panResponderAnimRef.current = new Animated.ValueXY({ x: 0, y: 0 });
-    return panResponderAnimRef.current;
-  };
-
-  if (icon === undefined) {
+    }
+
+    let backgroundColor = "";
     switch (type) {
-      case "success": {
-        if (successIcon) {
-          icon = successIcon;
-        }
-        break;
-      }
+        case "success":
+            backgroundColor = successColor || "rgb(46, 125, 50)";
```

Comment:
hardcoded color values, they are not reusable, store them as constants or in a styles object

Reply:
this is only important for built in ToastType, we're using only custom ones so that does not apply anywhere, but left to provide extension if ever necessary

```js
toast.show("Task finished successfully", {
  type: "normal | success | warning | danger | custom",
  placement: "top | bottom",
  duration: 4000,
  offset: 30,
  animationType: "slide-in | zoom-in",
});
```

default Toast is of type normal and this type is replaced in CustomToastProvider.jsx, other types outside of custom are not used